src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -137,5 +144,20 @@ ArgumentListSyntax args
 				);
 			}
 		}
+
+		// Not an extension method because there may be more cases (e.g. in the
+		// future) and if more than this fix + its analyzer used this logic
+		// there could be undesirable coupling if we handled more cases.
+		internal static ArgumentListSyntax GetArgs( SyntaxNode syntax ) {
+			if ( syntax is InvocationExpressionSyntax invocation ) {

```

Comment:
That's cool, but I meant

```csharp
switch( syntax ) {
  case InvocationExpressionSyntax invocation:
    return invocation.ArgumentList;
  case ObjectCreationExpressionSyntax objectCreation:
    return objectCreation.ArgumentList;
  default:
    return null;
}
```

Reply:
Whoa cool tech

(I wish you could have sealed enum types so `default` wasn't necessary.)